mesheryctl/pkg/utils/platform.go

```diff
@@ -21,6 +22,62 @@ var (
 	ManifestsFolder = "manifests"
 )
 
+type YamlStruct struct {
+	Contexts struct {
+		Local struct {
+			Endpoint string `yaml:"endpoint,omitempty"`
+    		Token string `yaml:"token,omitempty"`
+    		Platform string `yaml:"platform,omitempty"`
+    		Adapters struct {
+				MesheryIstio string `yaml:"meshery-istio,omitempty"` 
+    			MesheryLinkerd string `yaml:"meshery-linkerd,omitempty"`
+    			MesheryConsul string `yaml:"meshery-consul,omitempty"`
+    			MesheryNsm string `yaml:"meshery-nsm,omitempty"`
+    			MesheryKuma string `yaml:"meshery-kuma,omitempty"`
+    			MesheryCpx string `yaml:"meshery-cpx,omitempty"`
+    			MesheryOsm string `yaml:"meshery-osm,omitempty"`
+    			MesheryTraefikMesh string `yaml:"meshery-traefik-mesh,omitempty"`
+			} `yaml:"adapters,omitempty"`
+			Channel string `yaml:"channel,omitempty"`
+			Version string `yaml:"version,omitempty"`
+		} `yaml:"local,omitempty"`
+		Currentcontext string `yaml:"current-context,omitempty"`
+		Tokens struct {
+			Name string `yaml:"name,omitempty"`
+  			Location string `yaml:"location,omitempty"`
+		} `yaml:"tokens,omitempty"`
+	} `yaml:"contexts,omitempty"`
+	
+}
+//get cached k8s manifests
+func GetCachedManifests(version string) ([]Manifest, error) {
+	var yamlStruct YamlStruct
+	var localManifestList []Manifest
+
+	yamlFile, err := ioutil.ReadFile(filepath.Join(MesheryFolder, "config.yaml"))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read config.yaml file")
+	}
+
+	err = yaml.Unmarshal(yamlFile, &yamlStruct)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal config.yaml")
+	}
+
+	if version == yamlStruct.Contexts.Local.Version {
```

Comment:
I guess here, your approach is to compare the `version` from `current-context` with the version from cached manifests, but the version you're sending as a parameter and the one being fetched as `yamlStruct.Contexts.Local.Version` will be always equal because `YamlStruct` is nothing but unmarshalling `current-context`

Reply:
I am not sure I understand.  isn't version defined by the user as a flag in the cli?

Reply:
Nope, version is fetched from current context

Reply:
@piyushsingariya is there a flag that changes the version I should be looking out for?

Reply:
This is what needs to be done here, match the `version` from `current-context` with the version mentioned in each of the cached manifests files, and if that matches we deploy these cached manifests files. you can get some hints on how can you match the version from a manifest file from `ChangeManifestVersion()` from `platform.go`